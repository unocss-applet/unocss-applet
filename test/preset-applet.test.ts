import type { UnocssPluginContext } from '@unocss/core'
import { createGenerator } from '@unocss/core'
import presetApplet, { transformerApplet } from '@unocss-applet/preset-applet'
import { describe, expect, it } from 'vitest'
import MagicString from 'magic-string'

const targets = [
  // base
  'p-2',
  'p-2.5',
  'p-2.5rem',
  'p-2.5px',
  'p-2.5em',

  'b-2',
  'b-2.5',
  'b-2.5rem',
  'b-2.5px',
  'b-2.5em',

  // variants - mix
  'mix-tint-50-c-red-400',
  'mix-shade-50-c-red-400',
  'mix-shift-50-c-red-600',
  'mix-shift--50-c-red-600',

  // custom colors
  'text-custom-a',
  'bg-custom-b',
  'bg-info',
  'bg-info/10',
  'bg-info/[10%]',
  'border-custom-b',
  'border-custom-b/0',
  'border-custom-b/10',
  'bg-custom-c',
  'bg-custom-c/10',
  'bg-custom-d',
  'bg-custom-d/20',
  'bg-custom-e',
  'bg-custom-e/30',
  'bg-custom-f',
  'bg-custom-f/30',
  'bg-custom-f/[var(--f-op)]',

  // wind - placeholder
  'placeholder-red-400',
  'placeholder-inherit',
  'placeholder-opacity-10',
  'placeholder-op90',

  // wind - variants custom media (themed)
  'media-opacity_not_ok:opacity-0',
  'media-touch:p-4',

  // mini + wind - placeholder
  'focus:placeholder-red-300',
  'hover:placeholder-op90',

  // space
  'space-y-none',
  'space-x-2',
  'space-y-4',
  'space-x-reverse',
  'space-x-$space',
  'space-inline-2',
  'space-block-4',
  'space-block-none',
  'space-inline-reverse',
  'space-inline-$space',

  // divide
  'divide',
  'divide-y-4',
  'divide-x-4',
  'divide-x-reverse',
  'divide-block-4',
  'divide-inline-4',
  'divide-inline-reverse',
  'divide-green-500',
  'divide-opacity-50',
  'divide-dashed',
  'divide-dotted',
  'divide-ridge',
  'divide-transparent',
  'divide-current',
  'divide-none',
  'divide-x-none',
  'divide-inline-none',

  // media
  'md:divide-y-4',
  'md:bg-red',

  // shortcuts
  'u-text-color',
  'u-text-color/2',
]

const targets2 = [
  // mini - variants selector
  'selector-[section]:c-gray-400',
  'selector-[.cls.multi]:c-gray-400',
  'md:selector-[aside]:shadow-xl',
  'dark:selector-[.body_main]:bg-white',
  '[&_.nut-button]:text-lg',
  '[&_.nut-form-item\\_\\_label]:!text-left',
  '*:p-2',
  'dark:*:p-2',
]

const presetExtras = [
  'size-0',
  'size-1',
  'size-2',
  'size-0.5',
  'size-1.5',
  'size-1/2',
  'size-auto',
  'size-sm',
  'size-prose',
  'size-screen',
  'size-full',

  'min-size-0',
  'min-size-1',
  'min-size-2',
  'min-size-0.5',
  'min-size-1.5',
  'min-size-1/2',
  'min-size-auto',
  'min-size-sm',
  'min-size-prose',
  'min-size-screen',
  'min-size-full',

  'max-size-0',
  'max-size-1',
  'max-size-2',
  'max-size-0.5',
  'max-size-1.5',
  'max-size-1/2',
  'max-size-auto',
  'max-size-sm',
  'max-size-prose',
  'max-size-screen',
  'max-size-full',

  '[size~="\\30 20"]',
]

const uno = createGenerator({
  presets: [
    presetApplet({
      dark: 'media',
      unsupportedChars: ['~', ' '],
    }),
  ],
  theme: {
    colors: {
      custom: {
        a: 'var(--custom)',
        b: 'rgba(var(--custom), %alpha)',
        c: 'rgba(var(--custom-c) / %alpha)',
        d: 'hsl(var(--custom-d), %alpha)',
        e: 'hsl(var(--custom-e) / <alpha-value>)',
        f: 'lch(var(--custom-f) / <alpha-value>)',
      },
      info: 'hsl(200.1, 100%, 54.3%)',
    },
  },
  shortcuts: {
    'u-text-color': 'text-[#323233] dark:text-[#F5F5F5]',
    'u-text-color/2': 'text-[#969799] dark:text-[#707070]',
  },
})

describe('preset-applet', () => {
  it('targets', async () => {
    const code = targets.join(' ')
    const { css } = await uno.generate(code, { preflights: false })
    const { css: css2 } = await uno.generate(code, { preflights: false })

    expect(css).toMatchSnapshot()
    expect(css).toEqual(css2)
  })

  it('targets2', async () => {
    const code = targets2.join(' ')
    const { css } = await uno.generate(code, { preflights: false })
    const { css: css2 } = await uno.generate(code, { preflights: false })

    expect(css).toMatchSnapshot()
    expect(css).toEqual(css2)
  })

  it('preset extras', async () => {
    const code = presetExtras.join(' ')
    const { css } = await uno.generate(code, { preflights: false })

    expect(css).toMatchSnapshot()
  })
})

const transformer = transformerApplet()

describe('transformer-applet', () => {
  async function transform(code: string) {
    const s = new MagicString(code)
    await transformer.transform(s, '', {
      tokens: new Set<string>(),
      uno,
    } as UnocssPluginContext)
    return s.toString()
  }

  it('basic', async () => {
    const transformTargets = [
      '-ml-1.5 ml-1.5 -mt-2',
      'bg-[url(https://api.iconify.design/carbon:bat.svg?color=red)]',
    ]

    const result = await transform(transformTargets.join(' '))
    expect(result).toMatchInlineSnapshot(`"-ml-1_a_5 ml-1_a_5 -mt-2 bg-_a_url_a_https_a__a__a_api_a_iconify_a_design_a_carbon_a_bat_a_svg_a_color_a_red_a__a_"`)
  })
})
