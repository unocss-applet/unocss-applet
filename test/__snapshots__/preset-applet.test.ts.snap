// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`preset-applet > preset extras 1`] = `
"/* layer: default */
.size-0{width:0;height:0;}
.size-0_a_5{width:0.125rem;height:0.125rem;}
.size-1{width:0.25rem;height:0.25rem;}
.size-1_a_2{width:50%;height:50%;}
.size-1_a_5{width:0.375rem;height:0.375rem;}
.size-2{width:0.5rem;height:0.5rem;}
.size-auto{width:auto;height:auto;}
.size-full{width:100%;height:100%;}
.size-prose{width:65ch;height:65ch;}
.size-screen{width:100vw;height:100vh;}
.size-sm{width:24rem;height:24rem;}"
`;

exports[`preset-applet > targets 1`] = `
"/* layer: shortcuts */
.u-text-color{--un-text-opacity:1;color:rgb(50 50 51 / var(--un-text-opacity));}
.u-text-color_a_2{--un-text-opacity:1;color:rgb(150 151 153 / var(--un-text-opacity));}
@media (prefers-color-scheme: dark){
.u-text-color{--un-text-opacity:1;color:rgb(245 245 245 / var(--un-text-opacity));}
.u-text-color_a_2{--un-text-opacity:1;color:rgb(112 112 112 / var(--un-text-opacity));}
}
/* layer: default */
.space-x-_a_space>view+view,.space-x-_a_space>view+button,.space-x-_a_space>view+text,.space-x-_a_space>view+image,.space-x-_a_space>button+view,.space-x-_a_space>button+button,.space-x-_a_space>button+text,.space-x-_a_space>button+image,.space-x-_a_space>text+view,.space-x-_a_space>text+button,.space-x-_a_space>text+text,.space-x-_a_space>text+image,.space-x-_a_space>image+view,.space-x-_a_space>image+button,.space-x-_a_space>image+text,.space-x-_a_space>image+image{--un-space-x-reverse:0;margin-left:calc(var(--space) * calc(1 - var(--un-space-x-reverse)));margin-right:calc(var(--space) * var(--un-space-x-reverse));}
.space-x-2>view+view,.space-x-2>view+button,.space-x-2>view+text,.space-x-2>view+image,.space-x-2>button+view,.space-x-2>button+button,.space-x-2>button+text,.space-x-2>button+image,.space-x-2>text+view,.space-x-2>text+button,.space-x-2>text+text,.space-x-2>text+image,.space-x-2>image+view,.space-x-2>image+button,.space-x-2>image+text,.space-x-2>image+image{--un-space-x-reverse:0;margin-left:calc(0.5rem * calc(1 - var(--un-space-x-reverse)));margin-right:calc(0.5rem * var(--un-space-x-reverse));}
.space-y-4>view+view,.space-y-4>view+button,.space-y-4>view+text,.space-y-4>view+image,.space-y-4>button+view,.space-y-4>button+button,.space-y-4>button+text,.space-y-4>button+image,.space-y-4>text+view,.space-y-4>text+button,.space-y-4>text+text,.space-y-4>text+image,.space-y-4>image+view,.space-y-4>image+button,.space-y-4>image+text,.space-y-4>image+image{--un-space-y-reverse:0;margin-top:calc(1rem * calc(1 - var(--un-space-y-reverse)));margin-bottom:calc(1rem * var(--un-space-y-reverse));}
.space-y-none>view+view,.space-y-none>view+button,.space-y-none>view+text,.space-y-none>view+image,.space-y-none>button+view,.space-y-none>button+button,.space-y-none>button+text,.space-y-none>button+image,.space-y-none>text+view,.space-y-none>text+button,.space-y-none>text+text,.space-y-none>text+image,.space-y-none>image+view,.space-y-none>image+button,.space-y-none>image+text,.space-y-none>image+image{--un-space-y-reverse:0;margin-top:calc(0px * calc(1 - var(--un-space-y-reverse)));margin-bottom:calc(0px * var(--un-space-y-reverse));}
.space-x-reverse>view+view,.space-x-reverse>view+button,.space-x-reverse>view+text,.space-x-reverse>view+image,.space-x-reverse>button+view,.space-x-reverse>button+button,.space-x-reverse>button+text,.space-x-reverse>button+image,.space-x-reverse>text+view,.space-x-reverse>text+button,.space-x-reverse>text+text,.space-x-reverse>text+image,.space-x-reverse>image+view,.space-x-reverse>image+button,.space-x-reverse>image+text,.space-x-reverse>image+image{--un-space-x-reverse:1;}
.space-block-4{--un-space-block-reverse:0;margin-block-start:calc(1rem * calc(1 - var(--un-space-block-reverse)));margin-block-end:calc(1rem * calc(1 - var(--un-space-block-reverse)));}
.space-block-none{--un-space-block-reverse:0;margin-block-start:calc(0px * calc(1 - var(--un-space-block-reverse)));margin-block-end:calc(0px * calc(1 - var(--un-space-block-reverse)));}
.space-inline-_a_space{--un-space-inline-reverse:0;margin-inline-start:calc(var(--space) * calc(1 - var(--un-space-inline-reverse)));margin-inline-end:calc(var(--space) * calc(1 - var(--un-space-inline-reverse)));}
.space-inline-2{--un-space-inline-reverse:0;margin-inline-start:calc(0.5rem * calc(1 - var(--un-space-inline-reverse)));margin-inline-end:calc(0.5rem * calc(1 - var(--un-space-inline-reverse)));}
.space-inline-reverse{--un-space-inline-reverse:1;}
.divide-x-4>view+view,.divide-x-4>view+button,.divide-x-4>view+text,.divide-x-4>view+image,.divide-x-4>button+view,.divide-x-4>button+button,.divide-x-4>button+text,.divide-x-4>button+image,.divide-x-4>text+view,.divide-x-4>text+button,.divide-x-4>text+text,.divide-x-4>text+image,.divide-x-4>image+view,.divide-x-4>image+button,.divide-x-4>image+text,.divide-x-4>image+image{--un-divide-x-reverse:0;border-left-width:calc(4px * calc(1 - var(--un-divide-x-reverse)));border-right-width:calc(4px * var(--un-divide-x-reverse));}
.divide-x-none>view+view,.divide-x-none>view+button,.divide-x-none>view+text,.divide-x-none>view+image,.divide-x-none>button+view,.divide-x-none>button+button,.divide-x-none>button+text,.divide-x-none>button+image,.divide-x-none>text+view,.divide-x-none>text+button,.divide-x-none>text+text,.divide-x-none>text+image,.divide-x-none>image+view,.divide-x-none>image+button,.divide-x-none>image+text,.divide-x-none>image+image{--un-divide-x-reverse:0;border-left-width:calc(0px * calc(1 - var(--un-divide-x-reverse)));border-right-width:calc(0px * var(--un-divide-x-reverse));}
.divide-y-4>view+view,.divide-y-4>view+button,.divide-y-4>view+text,.divide-y-4>view+image,.divide-y-4>button+view,.divide-y-4>button+button,.divide-y-4>button+text,.divide-y-4>button+image,.divide-y-4>text+view,.divide-y-4>text+button,.divide-y-4>text+text,.divide-y-4>text+image,.divide-y-4>image+view,.divide-y-4>image+button,.divide-y-4>image+text,.divide-y-4>image+image{--un-divide-y-reverse:0;border-top-width:calc(4px * calc(1 - var(--un-divide-y-reverse)));border-bottom-width:calc(4px * var(--un-divide-y-reverse));}
.divide-x-reverse>view+view,.divide-x-reverse>view+button,.divide-x-reverse>view+text,.divide-x-reverse>view+image,.divide-x-reverse>button+view,.divide-x-reverse>button+button,.divide-x-reverse>button+text,.divide-x-reverse>button+image,.divide-x-reverse>text+view,.divide-x-reverse>text+button,.divide-x-reverse>text+text,.divide-x-reverse>text+image,.divide-x-reverse>image+view,.divide-x-reverse>image+button,.divide-x-reverse>image+text,.divide-x-reverse>image+image{--un-divide-x-reverse:1;}
.divide-block-4>view+view,.divide-block-4>view+button,.divide-block-4>view+text,.divide-block-4>view+image,.divide-block-4>button+view,.divide-block-4>button+button,.divide-block-4>button+text,.divide-block-4>button+image,.divide-block-4>text+view,.divide-block-4>text+button,.divide-block-4>text+text,.divide-block-4>text+image,.divide-block-4>image+view,.divide-block-4>image+button,.divide-block-4>image+text,.divide-block-4>image+image{--un-divide-block-reverse:0;border-block-start-width:calc(4px * calc(1 - var(--un-divide-block-reverse)));border-block-end-width:calc(4px * calc(1 - var(--un-divide-block-reverse)));}
.divide-inline-4>view+view,.divide-inline-4>view+button,.divide-inline-4>view+text,.divide-inline-4>view+image,.divide-inline-4>button+view,.divide-inline-4>button+button,.divide-inline-4>button+text,.divide-inline-4>button+image,.divide-inline-4>text+view,.divide-inline-4>text+button,.divide-inline-4>text+text,.divide-inline-4>text+image,.divide-inline-4>image+view,.divide-inline-4>image+button,.divide-inline-4>image+text,.divide-inline-4>image+image{--un-divide-inline-reverse:0;border-inline-start-width:calc(4px * calc(1 - var(--un-divide-inline-reverse)));border-inline-end-width:calc(4px * calc(1 - var(--un-divide-inline-reverse)));}
.divide-inline-none>view+view,.divide-inline-none>view+button,.divide-inline-none>view+text,.divide-inline-none>view+image,.divide-inline-none>button+view,.divide-inline-none>button+button,.divide-inline-none>button+text,.divide-inline-none>button+image,.divide-inline-none>text+view,.divide-inline-none>text+button,.divide-inline-none>text+text,.divide-inline-none>text+image,.divide-inline-none>image+view,.divide-inline-none>image+button,.divide-inline-none>image+text,.divide-inline-none>image+image{--un-divide-inline-reverse:0;border-inline-start-width:calc(0px * calc(1 - var(--un-divide-inline-reverse)));border-inline-end-width:calc(0px * calc(1 - var(--un-divide-inline-reverse)));}
.divide-inline-reverse>view+view,.divide-inline-reverse>view+button,.divide-inline-reverse>view+text,.divide-inline-reverse>view+image,.divide-inline-reverse>button+view,.divide-inline-reverse>button+button,.divide-inline-reverse>button+text,.divide-inline-reverse>button+image,.divide-inline-reverse>text+view,.divide-inline-reverse>text+button,.divide-inline-reverse>text+text,.divide-inline-reverse>text+image,.divide-inline-reverse>image+view,.divide-inline-reverse>image+button,.divide-inline-reverse>image+text,.divide-inline-reverse>image+image{--un-divide-inline-reverse:1;}
.divide-current>view+view,.divide-current>view+button,.divide-current>view+text,.divide-current>view+image,.divide-current>button+view,.divide-current>button+button,.divide-current>button+text,.divide-current>button+image,.divide-current>text+view,.divide-current>text+button,.divide-current>text+text,.divide-current>text+image,.divide-current>image+view,.divide-current>image+button,.divide-current>image+text,.divide-current>image+image{border-color:currentColor;}
.divide-green-500>view+view,.divide-green-500>view+button,.divide-green-500>view+text,.divide-green-500>view+image,.divide-green-500>button+view,.divide-green-500>button+button,.divide-green-500>button+text,.divide-green-500>button+image,.divide-green-500>text+view,.divide-green-500>text+button,.divide-green-500>text+text,.divide-green-500>text+image,.divide-green-500>image+view,.divide-green-500>image+button,.divide-green-500>image+text,.divide-green-500>image+image{--un-divide-opacity:1;border-color:rgb(34 197 94 / var(--un-divide-opacity));}
.divide-transparent>view+view,.divide-transparent>view+button,.divide-transparent>view+text,.divide-transparent>view+image,.divide-transparent>button+view,.divide-transparent>button+button,.divide-transparent>button+text,.divide-transparent>button+image,.divide-transparent>text+view,.divide-transparent>text+button,.divide-transparent>text+text,.divide-transparent>text+image,.divide-transparent>image+view,.divide-transparent>image+button,.divide-transparent>image+text,.divide-transparent>image+image{border-color:transparent;}
.divide-opacity-50>view+view,.divide-opacity-50>view+button,.divide-opacity-50>view+text,.divide-opacity-50>view+image,.divide-opacity-50>button+view,.divide-opacity-50>button+button,.divide-opacity-50>button+text,.divide-opacity-50>button+image,.divide-opacity-50>text+view,.divide-opacity-50>text+button,.divide-opacity-50>text+text,.divide-opacity-50>text+image,.divide-opacity-50>image+view,.divide-opacity-50>image+button,.divide-opacity-50>image+text,.divide-opacity-50>image+image{--un-divide-opacity:0.5;}
.divide-dashed>view+view,.divide-dashed>view+button,.divide-dashed>view+text,.divide-dashed>view+image,.divide-dashed>button+view,.divide-dashed>button+button,.divide-dashed>button+text,.divide-dashed>button+image,.divide-dashed>text+view,.divide-dashed>text+button,.divide-dashed>text+text,.divide-dashed>text+image,.divide-dashed>image+view,.divide-dashed>image+button,.divide-dashed>image+text,.divide-dashed>image+image{border-style:dashed;}
.divide-dotted>view+view,.divide-dotted>view+button,.divide-dotted>view+text,.divide-dotted>view+image,.divide-dotted>button+view,.divide-dotted>button+button,.divide-dotted>button+text,.divide-dotted>button+image,.divide-dotted>text+view,.divide-dotted>text+button,.divide-dotted>text+text,.divide-dotted>text+image,.divide-dotted>image+view,.divide-dotted>image+button,.divide-dotted>image+text,.divide-dotted>image+image{border-style:dotted;}
.divide-none>view+view,.divide-none>view+button,.divide-none>view+text,.divide-none>view+image,.divide-none>button+view,.divide-none>button+button,.divide-none>button+text,.divide-none>button+image,.divide-none>text+view,.divide-none>text+button,.divide-none>text+text,.divide-none>text+image,.divide-none>image+view,.divide-none>image+button,.divide-none>image+text,.divide-none>image+image{border-style:none;}
.divide-ridge>view+view,.divide-ridge>view+button,.divide-ridge>view+text,.divide-ridge>view+image,.divide-ridge>button+view,.divide-ridge>button+button,.divide-ridge>button+text,.divide-ridge>button+image,.divide-ridge>text+view,.divide-ridge>text+button,.divide-ridge>text+text,.divide-ridge>text+image,.divide-ridge>image+view,.divide-ridge>image+button,.divide-ridge>image+text,.divide-ridge>image+image{border-style:ridge;}
.b-2{border-width:2px;}
.b-2_a_5,
.b-2_a_5px{border-width:2.5px;}
.b-2_a_5em{border-width:2.5em;}
.b-2_a_5rem{border-width:2.5rem;}
.border-custom-b{--un-border-opacity:1;border-color:rgba(var(--custom), var(--un-border-opacity));}
.border-custom-b_a_0{border-color:rgba(var(--custom), 0);}
.border-custom-b_a_10{border-color:rgba(var(--custom), 0.1);}
.bg-custom-b{--un-bg-opacity:1;background-color:rgba(var(--custom), var(--un-bg-opacity));}
.bg-custom-c{--un-bg-opacity:1;background-color:rgba(var(--custom-c), var(--un-bg-opacity));}
.bg-custom-c_a_10{background-color:rgba(var(--custom-c), 0.1);}
.bg-custom-d{--un-bg-opacity:1;background-color:hsl(var(--custom-d), var(--un-bg-opacity));}
.bg-custom-d_a_20{background-color:hsl(var(--custom-d), 0.2);}
.bg-custom-e{--un-bg-opacity:1;background-color:hsl(var(--custom-e) / var(--un-bg-opacity));}
.bg-custom-e_a_30{background-color:hsl(var(--custom-e) / 0.3);}
.bg-custom-f{--un-bg-opacity:1;background-color:lch(var(--custom-f) / var(--un-bg-opacity));}
.bg-custom-f_a__a_var_a_--f-op_a__a_{background-color:lch(var(--custom-f) / var(--f-op));}
.bg-custom-f_a_30{background-color:lch(var(--custom-f) / 0.3);}
.bg-info{--un-bg-opacity:1;background-color:hsl(200.1 100% 54.3% / var(--un-bg-opacity));}
.bg-info_a__a_10_a__a_{background-color:hsl(200.1 100% 54.3% / 10%);}
.bg-info_a_10{background-color:hsl(200.1 100% 54.3% / 0.1);}
.p-2{padding:0.5rem;}
.p-2_a_5{padding:0.625rem;}
.p-2_a_5em{padding:2.5em;}
.p-2_a_5px{padding:2.5px;}
.p-2_a_5rem{padding:2.5rem;}
.text-custom-a{color:var(--custom);}
.mix-shade-50-c-red-400{--un-text-opacity:1;color:rgb(calc(248 + (0 - 248) * 50 / 100) calc(113 + (0 - 113) * 50 / 100) calc(113 + (0 - 113) * 50 / 100) / calc(var(--un-text-opacity) + (1 - var(--un-text-opacity)) * 50 / 100));}
.mix-shift--50-c-red-600{--un-text-opacity:1;color:rgb(calc(220 + (255 - 220) * 50 / 100) calc(38 + (255 - 38) * 50 / 100) calc(38 + (255 - 38) * 50 / 100) / calc(var(--un-text-opacity) + (1 - var(--un-text-opacity)) * 50 / 100));}
.mix-shift-50-c-red-600{--un-text-opacity:1;color:rgb(calc(220 + (0 - 220) * 50 / 100) calc(38 + (0 - 38) * 50 / 100) calc(38 + (0 - 38) * 50 / 100) / calc(var(--un-text-opacity) + (1 - var(--un-text-opacity)) * 50 / 100));}
.mix-tint-50-c-red-400{--un-text-opacity:1;color:rgb(calc(248 + (255 - 248) * 50 / 100) calc(113 + (255 - 113) * 50 / 100) calc(113 + (255 - 113) * 50 / 100) / calc(var(--un-text-opacity) + (1 - var(--un-text-opacity)) * 50 / 100));}
.focus_a_placeholder-red-300:focus::placeholder{--un-placeholder-opacity:1;color:rgb(252 165 165 / var(--un-placeholder-opacity));}
.placeholder-inherit::placeholder{color:inherit;}
.placeholder-red-400::placeholder{--un-placeholder-opacity:1;color:rgb(248 113 113 / var(--un-placeholder-opacity));}
.hover_a_placeholder-op90:hover::placeholder{--un-placeholder-opacity:0.9;}
.placeholder-op90::placeholder{--un-placeholder-opacity:0.9;}
.placeholder-opacity-10::placeholder{--un-placeholder-opacity:0.1;}
@media (hover: none) and (pointer: coarse){
.media-touch_a_p-4{padding:1rem;}
}
@media (prefers-reduced-transparency: reduce){
.media-opacity_not_ok_a_opacity-0{opacity:0;}
}
@media (min-width: 768px){
.md_a_divide-y-4>view+view,.md_a_divide-y-4>view+button,.md_a_divide-y-4>view+text,.md_a_divide-y-4>view+image,.md_a_divide-y-4>button+view,.md_a_divide-y-4>button+button,.md_a_divide-y-4>button+text,.md_a_divide-y-4>button+image,.md_a_divide-y-4>text+view,.md_a_divide-y-4>text+button,.md_a_divide-y-4>text+text,.md_a_divide-y-4>text+image,.md_a_divide-y-4>image+view,.md_a_divide-y-4>image+button,.md_a_divide-y-4>image+text,.md_a_divide-y-4>image+image{--un-divide-y-reverse:0;border-top-width:calc(4px * calc(1 - var(--un-divide-y-reverse)));border-bottom-width:calc(4px * var(--un-divide-y-reverse));}
.md_a_bg-red{--un-bg-opacity:1;background-color:rgb(248 113 113 / var(--un-bg-opacity));}
}"
`;

exports[`preset-applet > targets2 1`] = `
"/* layer: default */
._a__a_p-2>view,._a__a_p-2>button,._a__a_p-2>text,._a__a_p-2>image{padding:0.5rem;}
._a__a___a_nut-form-item_a___a__label_a__a__a_text-left .nut-form-item__label{text-align:left !important;}
._a__a___a_nut-button_a__a_text-lg .nut-button{font-size:1.125rem;line-height:1.75rem;}
.cls.multi,
section{--un-text-opacity:1;color:rgb(156 163 175 / var(--un-text-opacity));}
@media (prefers-color-scheme: dark){
.body main{--un-bg-opacity:1;background-color:rgb(255 255 255 / var(--un-bg-opacity));}
.dark_a__a__a_p-2>view,.dark_a__a__a_p-2>button,.dark_a__a__a_p-2>text,.dark_a__a__a_p-2>image{padding:0.5rem;}
}
@media (min-width: 768px){
aside{--un-shadow:var(--un-shadow-inset) 0 20px 25px -5px var(--un-shadow-color, rgb(0 0 0 / 0.1)),var(--un-shadow-inset) 0 8px 10px -6px var(--un-shadow-color, rgb(0 0 0 / 0.1));box-shadow:var(--un-ring-offset-shadow), var(--un-ring-shadow), var(--un-shadow);}
}"
`;
